docs/DeveloperGuide.adoc

```diff
@@ -12,255 +12,359 @@ ifdef::env-github[]
+Any data/state changes in `Model` will then propagate back to `View` automatically through https://docs.oracle.com/javafx/2/binding/jfxpub-binding.htm[JavaFX's Property and Binding].
 
-* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
-* At shut down: Shuts down the components and invokes cleanup method where necessary.
+In short, the App's data flow can be summarized as: +
+`View` -> `Logic` -> `Model` + `Storage` -> `View`
 
-<<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.
-The following class plays an important role at the architecture level:
+==== Architecture-level components
 
-* `LogsCenter` : Used by many classes to write log messages to the App's log file.
-
-The rest of the App consists of four components.
+Overall, the App consists of four components:
 
-* <<Design-Ui,*`UI`*>>: The UI of the App.
-* <<Design-Logic,*`Logic`*>>: The command executor.
-* <<Design-Model,*`Model`*>>: Holds the data of the App in-memory.
+* <<Design-View,*`View`*>>: View of the App.
+* <<Design-Logic,*`Logic`*>>: Business logic of the App.
+* <<Design-Model,*`Model`*>>: In-memory representation of the App's data/state.
 * <<Design-Storage,*`Storage`*>>: Reads data from, and writes data to, the hard disk.
 
-Each of the four components
+In addition, <<Design-Commons,*`Commons`*>> represents a collection of classes used by multiple other components.
+The following class plays an important role at the architecture level:
+
+* `LogsCenter` : Used by many classes to write log messages to the App's log file.
+
+Each of the four components:
 
 * Defines its _API_ in an `interface` with the same name as the Component.
 * Exposes its functionality using a `{Component Name}Manager` class.
 
-For example, the `Logic` component (see the class diagram given below) defines it's API in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class.
+For example, the `Logic` component defines it's API in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class.
 
-.Class Diagram of the Logic Component
-image::LogicClassDiagram.png[]
-
-[discrete]
 ==== How the architecture components interact with each other
 
-The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `new -t Notably -b Lorem ipsum`.
 
-.Component interactions for `delete 1` command
+.Component interactions for the `new -t Notably -b Lorem ipsum` command
 image::ArchitectureSequenceDiagram.png[]
 
-The sections below give more details of each component.
-
-[[Design-Ui]]
-=== UI component
+[[Design-View]]
+=== View component
 
-.Structure of the UI Component
+.Structure of the View Component
```

Comment:
The lines pointing from SuggestionsList, SideBarTree and BlockContent seem very congested and they overlap a lot, making it seem confusing. Can try to separate them!